pkg/kn/commands/service/service_list.go

```diff
@@ -61,6 +62,10 @@ func NewServiceListCommand(p *commands.KnParams) *cobra.Command {
 				return err
 			}
 
+			//sort serviceList by name
+			sorter := NewServicesSorter(serviceList)
+			sort.Sort(sorter)
```

Comment:
If we are to implement this sorting via `sort.Sort`, I'd recommend renaming to re-present the sorting key, for example:
```
sort.Sort(ByServiceName[serviceList])
```

makes the operation readable.

or, we could simply use this one liner

```suggestion
			sort.SliceStable(serviceList.Items, func(i, j int) bool { return serviceList.Items[i].Name < serviceList.Items[j].Name })
```

Reply:
I like having the named sorter, and I agree with your naming, Navid.

One thing to make it simpler and more idiomatic is to use the pattern the sort library built into golang does: instead of making a struct with a ServiceList in it, make a typedef for ServiceList with the appropriate methods:

Example from golang source https://golang.org/src/sort/sort.go?s=7078:7105#L270

```
type ByServiceName *v1alpha1.ServiceList

func (s ByServiceName) Len() int { ... }
```
etc

Reply:
Thanks for the suggestion. I used Navid's suggested change, because it looks much simpler.

Reply:
Please review again.